providers/implementations/ciphers/cipher_sm4_ccm.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/* Dispatch functions for SM4 CCM mode */
+
+#include "cipher_sm4_ccm.h"
+#include "prov/implementations.h"
+#include "prov/providercommon.h"
+
+static OSSL_FUNC_cipher_freectx_fn sm4_ccm_freectx;
```

Comment:
add:
```c
static OSSL_FUNC_cipher_newctx_fn sm4_ccm_newctx;
```
here.

Reply:
It looks like the existing declaration was removed, not the new one added.
GCM also needs these.

Reply:
This looks like a function pointer variable definition, not a prototype declaration of the function.

Reply:
It isn't.  It's ensuring that the function has the correct prototype.

Reply:
Thanks, I got it. The function prototype is not declared in DISPATCH.

Reply:
> add:
> 
> ```c
> static OSSL_FUNC_cipher_newctx_fn sm4_ccm_newctx;
> ```
> 
> here.

`sm4_ccm_newctx()` is just a function that is internal called, there is no need to add a extra declaration, the real function in core dispatch has been declared in the macro `IMPLEMENT_aead_cipher`.